### PR TITLE
Stop changing attribute_code in memory

### PR DESCRIPTION
**Pull request by midlan**
### Description (*)
We found code in magento thats changing attribute code in memory

### Manual testing scenarios (*)
Breaks configurable product edit if there is later access to attribute code. E.g.:
```php
$sku = Mage::getResourceModel('catalog/product')->getAttributeRawValue($product_id, "sku", $store_id);
```

Ends with:
`Column not found: 1054 Unknown column 'catalog_product_entity.simple_product_sku' in 'field list'`

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by tmotyl** [comment]
Have you checked why magento is doing so? is this custom attribute name used for something? Is it used to prevent mixing simple prod attributes with configurable ones?

**Comment by midlan** [comment]
No, I didn't find any reason.

**Review by Sekiphp — DISMISSED** [review]
seems ok

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Super/Config/Simple.php [hunk 1]
```diff
@@ -75,7 +75,9 @@ protected function _prepareForm()
                 if (!in_array($inputType, $availableTypes)) {
                     continue;
                 }
-                $attribute->setAttributeCode('simple_product_' . $attributeCode);
+
+                $attributeCode = $attribute->getAttributeCode();
+
```
Comment (fballiano) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Super/Config/Simple.php:80]:
```suggestion
```
not needed, the same code is a few line up

**Review by fballiano — APPROVED** [review]
rebased, tested and it doesn't seem to create any problem

after 7 years we could merge it

**Comment by kyrena** [comment]
This comes from [Magento 1.5.1.0](https://github.com/OpenMage/magento-lts/blame/3b14b96cd955e31c492b2a83e9d5a8a1de912158/app/code/core/Mage/Adminhtml/Block/Catalog/Product/Edit/Tab/Super/Config/Simple.php#L78).
I read the changelog, but I didn't find the reason.
Testing.

**Comment by kyrena** [comment]
When you try to create a new simple product from associated products tab at store view level, checkboxes _use default value_ are not checked by default.

**Comment by kyrena** [comment]
Not for create empty button.

before:



after:

**Comment by fballiano** [comment]
I didn't understand how to replicate, but you're saying that this doesn't happen without this PR?

**Comment by kyrena** [comment]
Yes, without the PR this problem isn't here.
A possible solution [1249-midlan.diff.txt](https://github.com/OpenMage/magento-lts/files/11768253/4-pull-1249-midlan.diff.txt)

**Comment by fballiano** [comment]
in this case I'd close this PR

**Comment by midlan** [comment]
@fballiano I'd give it second chance. I implemented proposed solution with `clone` from @kyrena in 05bbcccedeb16e73d81fe764cee87d544f146c3c
At this moment I don't have running Magento instance to test it on.

**Review by luigifab — APPROVED** [review]

**Pull request closed without merging**